webapp/components/survey/NodeDefDetails/store/actions/useCancelEdits.js

```diff
@@ -7,30 +7,44 @@ import { NodeDefsActions } from '@webapp/store/survey'
 
 import { State } from '../state'
 
-export const useCancelEdits = () => {
+export const useCancelEdits = ({ setState }) => {
   const history = useHistory()
   const dispatch = useDispatch()
 
   const cancelEdits = ({ state }) => async () => {
     const nodeDef = State.getNodeDef(state)
     const nodeDefOriginal = State.getNodeDefOriginal(state)
 
+    await setState(State.assocNodeDef(nodeDefOriginal))
+
     await dispatch(NodeDefsActions.cancelEdit({ nodeDef, nodeDefOriginal }))
 
     history.goBack()
   }
 
-  return useCallback(async ({ state }) => {
-    const dirty = State.isDirty(state)
-    if (dirty) {
-      dispatch(
-        DialogConfirmActions.showDialogConfirm({
-          key: 'common.cancelConfirm',
-          onOk: cancelEdits({ state }),
-        })
-      )
-    } else {
-      await cancelEdits({ state })()
-    }
-  }, [])
+  return useCallback(
+    async ({ state }) =>
+      new Promise((resolve, reject) => {
```

Comment:
is there a need to return new Promise here?

Reply:
see comment to line 43